addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionManager.java

```diff
@@ -0,0 +1,218 @@
+        assertStillUnderRetentionAfter(file, createRuleWithActions(RetentionRule.ApplicationPolicy.MANUAL,
+                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null, 1L, 0L, 0L, 0L, null, null), 1_000);
+    }
+
+    @Test
+    public void testManualImmediateRuleWithActions() throws InterruptedException {
+        RetentionRule testRule = createImmediateRuleMillis(RetentionRule.ApplicationPolicy.MANUAL, 100L,
+                null, Arrays.asList("Document.Trash"));
+
+        file = service.attachRule(file, testRule, session);
+        assertTrue(session.isRecord(file.getRef()));
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+
+        awaitRetentionExpiration(1000L);
+
+        file = session.getDocument(file.getRef());
+
+        // it has no retention anymore and trashed
+        assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
+        assertTrue(file.isTrashed());
+    }
+
+    @Test
+    public void testManualImmediateRule() throws InterruptedException {
+        RetentionRule testRule = createManualImmediateRuleMillis(100L);
+
+        file = service.attachRule(file, testRule, session);
+        assertTrue(file.isRecord());
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+        assertFalse(file.isLocked());
+
+        awaitRetentionExpiration(1000L);
+
+        file = session.getDocument(file.getRef());
+
+        // it has no retention anymore
+        assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
+    }
+
+    @Test
+    public void testManualDocumentMovedToFolderRule() throws InterruptedException {
+
+        RetentionRule testRule = createManualEventBasedRuleMillis(DocumentEventTypes.DOCUMENT_MOVED,
+                "document.getPathAsString().startsWith('/testFolder')", 1000L);
+
+        file = service.attachRule(file, testRule, session);
+        assertTrue(file.isRecord());
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+        Record record = file.getAdapter(Record.class);
+        assertTrue(record.isRetainUntilInderterminate());
+
+        awaitRetentionExpiration(500L);
+
+        file = session.getDocument(file.getRef());
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+        record = file.getAdapter(Record.class);
+        assertTrue(record.isRetainUntilInderterminate());
+
+        DocumentModel folder = session.createDocumentModel("/", "testFolder", "Folder");
+        folder = session.createDocument(folder);
+        folder = session.saveDocument(folder);
+
+        file = session.move(file.getRef(), folder.getRef(), null);
+
+        awaitRetentionExpiration(500L);
+
+        record = file.getAdapter(Record.class);
+        assertFalse(record.isRetainUntilInderterminate());
+        assertFalse(record.isRetentionExpired());
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+
+        awaitRetentionExpiration(500L);
+
```

Comment:
`awaitRetentionExpiration(500L);` -> `awaitRetentionExpiration(500);` for the three last use